File: python/396.Rotate-Function.py
class Solution:
    def maxRotateFunction(self, nums: List[int]) -> int:
        n = len(nums)
        total_sum = 0
        temp = 0

        for i in range(n):
            total_sum += nums[i]
            temp += i * nums[i]

        max_val = temp

        for k in range(1, n):
            temp = temp + total_sum - n * nums[n - k]
            max_val = max(max_val, temp)

        return max_val

File: javascript/396.Rotate-Function.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxRotateFunction = function (nums) {
  const n = nums.length;
  let totalSum = 0;
  let temp = 0;

  for (let i = 0; i < n; i++) {
    totalSum += nums[i];
    temp += i * nums[i];
  }

  let maxVal = temp;

  for (let k = 1; k < n; k++) {
    temp = temp + totalSum - n * nums[n - k];
    maxVal = Math.max(maxVal, temp);
  }

  return maxVal;
};
